index.js

```diff
@@ -71,7 +71,15 @@ function redisStore(args) {
       }
 
       if (opts.parse) {
-        result = JSON.parse(result);
+
+        try {
+          // allow undefined only if allowed by isCacheableValue
+          if(! ( (result === undefined || result === "undefined") && typeof args.isCacheableValue === 'function' && args.isCacheableValue(result))) {
```

Comment:
On which case result can come with `undefined` value? I think it can come only with `"undefined"` (non-existing key in redis returns `null`). About the isCacheable, a non-cacheable value should not be stored in the first place, and we should retrieve then `null` no?